app/tasks/page.tsx

```diff
@@ -4,32 +4,89 @@ import { TaskHeader } from "@/components/TaskHeader";
 import { TaskList } from "@/components/TaskList";
 import tasksData from "@/data/taskData.json";
 import { Task } from "@/app/types/tasks";
+import TodoForm from "@/components/TodoForm";
+import { FORM_MODE, TASK_STATUS } from "../constants/Task";
+
+
 
 const Tasks = () => {
   const [tasks, setTasks] = useState<Task[]>([]);
+  const [isCreateAndEditFormVisible, setIsCreateAndEditFormVisible] = useState(false)
+  const [activeTask, setActiveTask] = useState<Task | undefined>();
 
   useEffect(() => {
     setTasks(tasksData);
   }, []);
 
-  const todoTasks = tasks.filter((task) => task.status === "todo");
-  const inProgressTasks = tasks.filter((task) => task.status === "in-progress");
+
+  const handleTaskSelect = (task: Task) => {
+    setActiveTask(task);
+    setIsCreateAndEditFormVisible(false);
+  };
+
+  const todoTasks = tasks.filter((task) => task.status === TASK_STATUS.TODO);
+  const inProgressTasks = tasks.filter((task) => task.status === TASK_STATUS.IN_PROGRESS);
+
+  const handleCreateSubmit = (data: Task) => {
+    console.log("Creating todo:", data)
+  }
+
+  const handleAcknowledge = () => {
+    console.log("Todo acknowledged")
+  }
+
+
+  const handleFormClose = () => {
+    setIsCreateAndEditFormVisible(false);
+    setActiveTask(undefined);
+  };
```

Comment:
- NIT: let's arrange the contents of the components in the following format:
```tsx
const ComponentName = () => {
   ... hooks excluding useEffect (useState, useReducer, custom hooks)
   
   ...variables (example: todoTasks, inProgressTasks)
   
   ... functions (example: handleCreateSubmit, handleAcknowledge)
   
   ...useEffect
   
   return(
       ...jsx
   )
}
```

- this will help us maintain consistency across components.